share/mk/bsd.cpu.mk

```diff
@@ -416,15 +416,24 @@ MACHINE_ABI+=	soft-float
 .else
 MACHINE_ABI+=	hard-float
 .endif
-.if (${MACHINE_ARCH:Maarch64*c*} || ${MACHINE_ARCH:Mriscv*c*})
-MACHINE_ABI+=	purecap
-.endif
 # Currently all 64-bit architectures include 64 in their name (see arch(7)).
 .if ${MACHINE_ARCH:M*64*}
+MACHINE_ABI+=	long64
+.else
+MACHINE_ABI+=	long32
+.endif
+.if (${MACHINE_ARCH:Maarch64*c*} || ${MACHINE_ARCH:Mriscv*c*})
+MACHINE_ABI+=	purecap
+.elif ${MACHINE_ABI:Mlong64}
 MACHINE_ABI+=	ptr64
 .else
 MACHINE_ABI+=	ptr32
 .endif
+.if ${MACHINE_ARCH} == "i386"
+MACHINE_ABI+=	time32
+.else
+MACHINE_ABI+=	time64
+.endif
```

Comment:
It would be good to document somewhere, maybe at the top here as a comment, all possible MACHINE_ABI values and their meanings.

Reply:
+1